### ISSUE TITLE
[Request] Import as Real-time Facade

### ISSUE THREAD

**Issue by tonysm** [laravel-idea/plugin#74]
This idea came from a [Twitter thread](https://twitter.com/tonysmdev/status/1270465138370457603), and I think it would be nice if we had a way to import classes as [Real-Time Facades](https://laravel.com/docs/7.x/facades#real-time-facades).

Something like this:

![image](https://user-images.githubusercontent.com/1178621/85790071-05032380-b706-11ea-9f3d-a5126f214e3c.png)

That would import the class, but prefixing it with `Facades\` so it can be used as a real-time facade.

**Comment by adelf** [comment]
Hi, Tony.

It's possible, but what about these real time facades method completion?
Are you ok without it? Maybe create also some kind of real time facades
helper code generator? Like, create helper code for all App/Services/*
classes?

**Comment by tonysm** [comment]
There were also some discussions about adding some docblocks to the generated facade file, so we could have autocompletion (something like `@mixin`, but we would then have to map the methods to static methods on the facade somehow).

Here's where Laravel creates the file for the real-time [facade](https://github.com/laravel/framework/blob/7.x/src/Illuminate/Foundation/AliasLoader.php#L101-L112).

I would say it's ok without it. But what are the options here? The plugin would be the one creating the file?

The pain-point from the thread is mostly PHPStorm not aware that classes prefixed with `Facades\` namespace are dynamically generated. Autocompletion would be a plus (a very good one, though).

**Comment by adelf** [comment]
My proposal is a new code generation, like "Generate Eloquent helper code". Let say "Generate Realt-time Facades helper code". User can choose parent namespace there.  Like "App/Services" Or, just "App"(but there will be a lot of classes). Laravel Idea will generate helper code for each class:
```
namespace Facades\App\Services;

/**
 * @method static User create($name)
 * @method static void delete($id)
 */
class UserService{}
```

So, full autocomplete for all these facades. What do you think?

**Comment by tonysm** [comment]
That sounds really good to me. I would personally only generate it for a single class, as needed, not in bulk, but I guess it's up to the developer to decide what they want to do, so I like the solution! :smile:

**Comment by inxilpro** [comment]
Would it be possible to auto-generate the facade if it's referenced in a file?

**Comment by adelf** [comment]
@inxilpro what do you mean?

**Comment by inxilpro** [comment]
I'm not sure if this could happen—but what if when the plug-in encountered a real-time facade, it could auto-generate the file if it was missing?

**Comment by raveren** [comment]
Yeah, this would be a killer feature for me, full support for real time facades. 

I need to make a zero-tested legacy app testable, and I'm approaching the real-time facades as a potential solution to inject dependencies without rewriting anything. Unfortunately the IDE does not support me in my good fight - neither the `Facades\` prefixed namespace not the static usage of the real-time-facade'd methods are recognized.

**Comment by nicolasbuch** [comment]
+1 for real time facade support. 

`Generate helper code` does not currently generate helper code for real time facades. 

Methods are not recognised and the IDE cannot find declaration to go to due to the prepended `Facades\` namespace. 
e.g. `Facades\App\Services\Formatter`

@adelf since the "will be fixed/implemented soon" label was added in 2020, would you mind giving us an update? 😄

**Comment by adelf** [comment]
I can generate helper code for every class in the project, but it will be super slow. So, it's not an option.

However, there is a way... fully magical as real-time facades themselves) I can try.

**Comment by nicolasbuch** [comment]
>I can generate helper code for every class in the project, but it will be super slow. So, it's not an option.
>
>However, there is a way... fully magical as real-time facades themselves) I can try.

That would be amazing.

Currently, we have to use the barryvdh `ide-helper:generate` command to get working helper code, which kinda defeats the purpose of using the laravel-idea `Generate helper code`.

Appreciate you looking into it!

**Comment by adelf** [comment]
> However, there is a way... fully magical as real-time facades themselves) I can try.

It's implemented in the 9.0 version. RT facades work like usual classes.
